scipy/stats/tests/test_stats.py

```diff
@@ -4200,6 +4212,105 @@ def test_kurtosistest_too_few_samples():
+                  592.34083395416258, 448.73177590617018, 300.61495185038905,
+                  187.97508449019588])
+
+    alt_two = 'x and y are sampled from different populations'
+    alt_less = 'x is sampled from a population of smaller values than y'
+    alt_greater = 'x is sampled from a population of larger values than y'
+
+    # p-values and statistic compared against wilcox.test from R stats
+    def test_two_tailed_exact(self):
+        res = stats.stats.mann_whitney_u(self.a, self.b)
+        ra = [res.statistic, res.pvalue]
+        assert_array_almost_equal(ra, (16, 0.703571428571428625), decimal=12)
+        assert_string_equal(res.alternative, self.alt_two)
+
+    def test_less_exact(self):
+        res = stats.stats.mann_whitney_u(self.a, self.b, alternative='less')
+        ra = [res.statistic, res.pvalue]
+        assert_array_almost_equal(ra, (23, 0.694642857142857117), decimal=12)
+        assert_string_equal(res.alternative, self.alt_less)
+
+    def test_greater_exact(self):
+        res = stats.stats.mann_whitney_u(self.a, self.b, alternative='greater')
+        ra = [res.statistic, res.pvalue]
+        assert_array_almost_equal(ra, (16, 0.351785714285714313), decimal=12)
+        assert_string_equal(res.alternative, self.alt_greater)
+
+    def test_two_tailed_inverse_exact(self):
+        res = stats.stats.mann_whitney_u(self.b, self.a)
+        ra = [res.statistic, res.pvalue]
+        assert_array_almost_equal(ra, (16, 0.703571428571428625), decimal=12)
+        assert_string_equal(res.alternative, self.alt_two)
+
+    def test_two_tailed_approx(self):
+        res = stats.stats.mann_whitney_u(self.x, self.y)
+        ra = [res.statistic, res.pvalue]
+        assert_array_almost_equal(ra, (16980.5, 5.64286553126644867e-05),
+                                  decimal=12)
+        assert_string_equal(res.alternative, self.alt_two)
+
+    def test_less_approx(self):
+        res = stats.stats.mann_whitney_u(self.x, self.y, alternative='less')
+        ra = [res.statistic, res.pvalue]
+        assert_array_almost_equal(ra, (16980.5, 2.8214327656332243e-05),
+                                  decimal=12)
+        assert_string_equal(res.alternative, self.alt_less)
+
+    def test_greater_approx(self):
+        res = stats.stats.mann_whitney_u(self.x, self.y, alternative='greater')
+        ra = [res.statistic, res.pvalue]
+        assert_array_almost_equal(ra, (21571.5, 0.9999719954296038),
+                                  decimal=12)
+        assert_string_equal(res.alternative, self.alt_greater)
+
+    def test_two_tailed_inverse_approx(self):
+        res = stats.stats.mann_whitney_u(self.y, self.x)
+        ra = [res.statistic, res.pvalue]
+        assert_array_almost_equal(ra, (16980.5, 5.64286553126644867e-05),
+                                  decimal=12)
+        assert_string_equal(res.alternative, self.alt_two)
+
+
 class TestMannWhitneyU(object):
```

Comment:
This class contains the `mannwhitneyu` tests that are in master. It has been modified to perform the same tests on `mannwhitneyu` and `mannwhitneyu2` temporarily. Ultimately, the tests from here that we want to keep will be moved to `test_hypotest.py`.